BurgerQueen/src/app/@core/authentication/services/auth.service.spec.ts

```diff
@@ -1,53 +1,69 @@
 import { TestBed } from '@angular/core/testing';
 import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
-import { AuthService } from './auth.service';
-import { Credentials, LoginResponse } from '../../../shared/models/Login';
+import { AuthService, Credentials } from './auth.service';
+import { LoginResponse } from '../../../shared/models/Login';
+import { of } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { LocalStorageService } from '../../services/local-storage.service';
+import { HttpClient } from '@angular/common/http';
+
+class MockLocalStorageService {
+  setStorage() {}
+}
 
 describe('AuthService', () => {
   let authService: AuthService;
   let httpTestingController: HttpTestingController;
+  let httpClientSpy: { post: jasmine.Spy };
 
   beforeEach(() => {
+    httpClientSpy = jasmine.createSpyObj('HttpClient', ['post']);
+
     TestBed.configureTestingModule({
       imports: [HttpClientTestingModule],
-      providers: [AuthService]
-    });
+      providers: [
+        AuthService,
+        { provide: LocalStorageService, useClass: MockLocalStorageService },
+        { provide: HttpClient, useValue: httpClientSpy },
+      ],    });
+    
     authService = TestBed.inject(AuthService);
     httpTestingController = TestBed.inject(HttpTestingController);
   });
 
-  it('should be created', () => {
-    expect(authService).toBeTruthy();
-  });
 
-  it('should return a token and role when calling getToken and getRole', () => {
-    localStorage.setItem('accessToken', 'fakeAccessToken');
-    localStorage.setItem('role', 'admin');
+  it('send a POST request when calling login', () => {
+    const mockCredentials: Credentials = { email: 'test@test.com', password: 'password' };
 
-    const token = authService.getToken();
-    const role = authService.getRole();
+    const mockResponse :LoginResponse = {
+        user: {
+          id: 1,
+          role: 'user',
+          email: 'test@test.com',
+        },
+        accessToken: 'mockAccessToken',
+      };
 
-    expect(token).toBe('fakeAccessToken');
-    expect(role).toBe('admin');
-  });
+    httpClientSpy.post.and.returnValue(of(mockResponse));
 
-  it('should send a POST request to the login endpoint and return a response', () => {
-    const mockCredentials: Credentials = { email: 'user@prueba.com', password: 'password' };
-    const mockResponse: LoginResponse = { accessToken: 'asdtoken', user:{ email: "hola@hola", id: 1, role: "admin"} };
+    authService.login(mockCredentials);
 
-    authService.sigIn(mockCredentials).subscribe((response) => {
-      expect(response).toEqual(mockResponse);
+    expect(httpClientSpy.post).toHaveBeenCalledWith(`${environment.apiUrl}/login`, mockCredentials);
+    
+    authService.loginResponse$.subscribe((response) => {
+      expect(response.data).toEqual(mockResponse);
```

Comment:
ya existe una subcripcion a este obsrbbale en el servico lo que debemos validar son las consecuencias de ese de esa manera no validamos implemnetacion si no validamos funcionalidad 

`si pasa X entonces debe resultar Y`

Reply:
en otro test incorpore tu sugerencia anterior de corroborar que se llena correctamente systemUser y el localStorage , ahora en este caso me recomendarias que quite la parte del susccribe ? y que quede así este test  : 

`it('send a POST request when calling login', () => {
    httpClientSpy.post.and.returnValue(of(mockResponse));

    authService.login(mockCredentials);

    expect(httpClientSpy.post).toHaveBeenCalledWith(`${environment.apiUrl}/login`, mockCredentials);
    
  });`